### PR TITLE
added address geocode to getting state reps, office phone numbers

### PR DESCRIPTION
**Pull request by twblackwell**
TODO: fix frontend to make it closer to the Figma
TODO: get feedback from Leg team

### CODE REVIEW

File: frontend/src/components/Bill/Bill.jsx [hunk 1]
```diff
@@ -20,34 +30,180 @@ function getSponsorNumber(billData) {
...
+        </div>
+      </div>
+      
+      <div className="action">
+        <h2>Take Action!</h2>
+        <div>
+          <form onSubmit={handleAddressSubmit}>
+            <input
+              type="text"
+              placeholder="Enter full address:"
```
Comment (elliott-king) [inline frontend/src/components/Bill/Bill.jsx]: Only the ZIP code is needed.

File: frontend/src/components/Bill/Bill.jsx [hunk 2]
```diff
@@ -20,34 +30,180 @@ function getSponsorNumber(billData) {
...
+          {loading && <p>Loading representatives...</p>}
+          {error && <p>{error}</p>}
+          <div>
+            <h3>Your Representatives:</h3>
+            {repsData
+              .filter(rep => rep.jurisdiction.classification !== "country")
+              .map(rep => (
+                <div key={rep.id}>
+                  <h4>{rep.name} - {rep.current_role.title}</h4>
+                  <img src={rep.image} alt={rep.name} style={{ maxWidth: "200px" }} />
```
Comment (elliott-king) [inline frontend/src/components/Bill/Bill.jsx]: The images slow down the page, and they are inconsistent sizes. I think we should omit them.

File: frontend/src/components/Bill/Bill.jsx [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
-import {useState, useEffect} from "react";
+import { PropTypes } from "prop-types";
+import { useState, useEffect, useCallback } from "react";
 import { useParams } from "react-router-dom";
 
-const RelatedBills = ({related}) => {
+import './Bill.css';
+
+const MAPBOX_TOKEN = "sk.eyJ1IjoidHdibGFja3dlbGwiLCJhIjoiY2x0cWMzOWsxMDRtYjJrcXdmN2R1Zml3bSJ9.IABJE1MnM1zJeYkrGUGJ9g";
+const OPENSTATES_API_KEY = "2a046c1a-de61-4a68-8ad8-f6a3fd0c42f3";
```
Comment (elliott-king) [inline frontend/src/components/Bill/Bill.jsx:9]: API token handling is _incredibly important_ so please read all of this.

1. don't put an API key in code
This repo is public, so everybody can view your API key. There are bots that scan GH for API keys.

2. don't use an API key on the frontend (if possible). I can see your tokens on the frontend:
Screen Shot 2024-03-14 at 1 44 26 PM

`MAPBOX_TOKEN` could be used, but you would [use a URL restriction](https://docs.mapbox.com/help/troubleshooting/how-to-use-mapbox-securely/#access-tokens)
`OPENSTATES_API_KEY` has no URL restrictions, so it should be used backend-only.

[Here is a good article](https://support.google.com/googleapi/answer/6310037?hl=en)

With those things in mind, I think you should:
1. [backend] Make an API endpoint on the backend that takes a user's zip code (or address?).
2. [backend] Make these two requests to mapbox & openstates.
3. [backend] Backend returns the resulting JSON for the user's legislator
4. [frontend] Frontend makes one API call `/your/new/api/endpoint?zip=<user-provided-zip>`

The user flow appears the same, but the API keys are used on the backend only.

In summation:
- ☑️ move 3rd-party API calls to backend
- ☑️ go to mapbox & openstates and reset both of your API keys

DM me if you want to chat about this.

**Review by elliott-king — CHANGES_REQUESTED** [review]

File: frontend/src/components/Bill/Bill.jsx (hunk 3 above)
Comment (twblackwell) [reply frontend/src/components/Bill/Bill.jsx:9]: totally agreed - oversight by me last night that I forgot to mention and fix this morning. I'll try to follow the pattern used for backend keys for sure

File: frontend/src/components/Bill/Bill.jsx (hunk 2 above)
Comment (twblackwell) [reply frontend/src/components/Bill/Bill.jsx:166]: removed! I think it looked nice, but agree that it slows it down, good idea!

Comment (elliott-king) [reply frontend/src/components/Bill/Bill.jsx:166]: There are specific member pages (not fleshed out). We could put images there.

**Comment by elliott-king** [comment]
This is currently on the `Bill` page, but might make more sense as part of the header. Then it will "follow" the user between pages. They could click it to go the the legislator's page.

**Comment by twblackwell** [comment]
Removed the conflicts since it makes sense to start over the user's local reps to be in the header instead, to keep it consistent. Will add back in the frontend items once I get a cleaner pull first

**Pull request closed without merging**